csrc/flash_attn/src/flash_fwd_kernel.h

```diff
@@ -828,8 +833,8 @@ inline __device__ void compute_attn_1rowblock_splitkv(const Params &params, cons
             flash::copy_w_min_idx<Is_even_K>(
                 tVgVnew, tVgV, tKVcKV, tKVpKV, binfo.actual_seqlen_k - n_block * kBlockN, binfo.seqlen_k_cache - n_block * kBlockN
             );
-            tVgV.data() = tVgV.data() + (-int(kBlockN * params.v_row_stride));
-            tVgVnew.data() = tVgVnew.data() + (-int(kBlockN * params.vnew_row_stride));
+            tVgV.data() = tVgV.data() + binfo.k_advance_offset_pg(bidb_cache, n_block, params.v_row_stride, kBlockN);
```

Comment:
this and the next change copies data from the end of the blocks, block by block (k_advance_offset_pg moves the pointer as much as a single flash attention block) from knew -> k and vnew -> v